### PR TITLE
Update examples for hosting PSCore and remove old outdated examples before 6.0.0 RTM

### PR DESCRIPTION
**Pull request by bergmeister**
## PR Summary

Update project and NuGet packages to `netcoreapp2.1` and `6.1.1` NuGet packages, old projects before 6.0 RTM time are being removed.
I checked it works by running `dotnet run` on the project.
cc @rkeithhill 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: docs/host-powershell/README.md [hunk 1]
```diff
@@ -2,155 +2,23 @@
...
-    }
-}
-```
-
-[CorePsAssemblyLoadContext.cs]: https://github.com/PowerShell/PowerShell/blob/v6.0.0-beta.2/src/Microsoft.PowerShell.CoreCLR.AssemblyLoadContext/CoreCLR/CorePsAssemblyLoadContext.cs
-[Resolving]: https://github.com/dotnet/corefx/blob/ec2a6190efa743ab600317f44d757433e44e859b/src/System.Runtime.Loader/ref/System.Runtime.Loader.cs#L35
-[bootstrapping]: https://github.com/PowerShell/PowerShell/blob/v6.0.0-beta.2/src/powershell/Program.cs#L27
+Please see the [.NET Core Sample Application](#net-core-sample-application) section for an example that uses PowerShell Core NuGet packages.
 
+[CorePsAssemblyLoadContext.cs]: https://docs.microsoft.com/dotnet/api/system.management.automation.powershellassemblyloadcontextinitializer.setpowershellassemblyloadcontext
```
Comment (iSazonov) [inline docs/host-powershell/README.md]: Is it still used?

File: docs/host-powershell/README.md [hunk 2]
```diff
@@ -2,155 +2,23 @@
...
+[Resolving]: https://github.com/dotnet/corefx/blob/d6678e9653defe3cdfff26b2ff62135b6b22c77f/src/System.Runtime.Loader/ref/System.Runtime.Loader.cs#L38
```
Comment (iSazonov) [inline docs/host-powershell/README.md]: Is it still used?

File: docs/host-powershell/README.md (hunk 1 above)
Comment (daxian-dbw) [reply docs/host-powershell/README.md]: Yes, it's still used by the DSC native host that is used in Azure. The DSC host doesn't have the powershell assemblies in its TPA list, and thus is loading powershell in a different way.

Comment (daxian-dbw) [reply docs/host-powershell/README.md]: Well, I think I misunderstood your comment :) No, I don't find this link used anywhere in this doc after the change.

Comment (daxian-dbw) [reply docs/host-powershell/README.md]: But we need to keep the introduction about `SetPowerShellAssemblyLoadContext` as it's needed for a special hosting scenario.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: docs/host-powershell/README.md (hunk 1 above)
Comment (bergmeister) [reply docs/host-powershell/README.md]: Ok, I brought those docs back in [this](https://github.com/PowerShell/PowerShell/pull/8472/commits/98551c83be5cdf737dca4ac72b494ac9142b4c44) self-contained commit

File: docs/host-powershell/README.md [hunk 3]
```diff
@@ -2,155 +2,23 @@
...
-- [sample-dotnet1.1](./sample-dotnet1.1) - .NET Core `1.1` + PowerShell Core `alpha.17` NuGet packages.
-  [.NET Core SDK 1.0.1](https://github.com/dotnet/cli/releases/tag/v1.0.1) is required.
-- [sample-dotnet2.0-powershell.beta.1](./sample-dotnet2.0-powershell.beta.1) - .NET Core `2.0.0` + PowerShell Core `beta.1` NuGet packages.
-  .NET Core SDK `2.0.0-preview1-005952` or higher is required.
-- [sample-dotnet2.0-powershell.beta.3](./sample-dotnet2.0-powershell.beta.3) - .NET Core `2.0.0` + PowerShell Core `beta.3` NuGet packages.
-  .NET Core SDK `2.0.0-preview1-005952` or higher is required.
-- [sample-dotnet2.0-powershell-crossplatform](./sample-dotnet2.0-powershell-crossplatform) - .Net Core `2.0.0` + PowerShell Core `6.0.1.1` NuGet packages.
+Note: The .NET Core `2.1` runtime and .NET Core SDK `2.1` or higher is required for the examples below:
+
+- [sample-windows](./sample-windows)
```
[the remark this answers is not in the document]
Comment (bergmeister) [reply docs/host-powershell/README.md]: Ok, I merged them together in [this](https://github.com/PowerShell/PowerShell/pull/8472/commits/54421590976dd9e3e31221454114cf80cd670514) self-contained commit

File: docs/host-powershell/README.md [hunk 4]
```diff
@@ -2,155 +2,23 @@
...
+- [sample-crossplatform](./sample-crossplatform)
 
-You can find the sample application project `"MyApp"` in each of the above 3 sample folders.
+You can find the sample application project `MyApp` in each of the above 2 sample folders.
 To build the sample project, run the following commands (make sure the required .NET Core SDK is in use):
 
 ```powershell
-dotnet restore .\MyApp\MyApp.csproj
 dotnet publish .\MyApp -c release -r win10-x64
 ```
```
[the remark this answers is not in the document]
Comment (bergmeister) [reply docs/host-powershell/README.md]: Yes, I might a slight adaption to not specify the runtime explicitly but rather specify the runtime in the csproj file.

**Comment by bergmeister** [comment]
@daxian-dbw I addressed your comments, the PR is up for re-review

**Review by daxian-dbw — APPROVED** [review]
Thanks @bergmeister for the updates!
I updated the `Special Hosting Scenario` section of the readme. Everything else looks great!

**Pull request merged**